hswidget/cof_state.go

```diff
@@ -0,0 +1,54 @@
+package hswidget
+
+import "github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2cof"
+
+type cofEditorMode int
+
+const (
+	cofEditorModeViewer cofEditorMode = iota
+	cofEditorModeAddLayer
+	cofEditorModeConfirm
+)
+
+// COFState represents cof editor's and viewer's state
+type COFState struct {
+	*viewerState
+	*newLayerFields
+	mode cofEditorMode
+}
+
+// Dispose clear widget's state
+func (s *COFState) Dispose() {
+	s.viewerState.Dispose()
+	s.newLayerFields.Dispose()
+}
+
+// viewerState represents cof viewer's state
+type viewerState struct {
```

Comment:
should be `COFViewerState`, because we'll have more then one structure, which describes viewers' states in this package

Reply:
oh, sorry, was planning on moving all of this into its own package

Reply:
will resolve in upcoming PR

Reply:
I've decided to resolve this in this PR since i'm already refactoring cof editor+widget. in 023fd2f, i've made a sub-package `hswidget/cofwidget/`